test/test_distribution_generic.py

```diff
@@ -46,15 +58,15 @@ def __getattribute__(self, attr):
 
 class DistTestCase:
 
-    def __init__(self, raw_dist, raw_params, expected_value_domain):
+    def __init__(self, raw_dist, raw_params, expected_value_domain, xfail=False):
         self.raw_dist = raw_dist
         self.raw_params = raw_params
         self.expected_value_domain = expected_value_domain
         for name, raw_param in self.raw_params:
             if get_backend() != "numpy":
                 # we need direct access to these tensors for gradient tests
                 setattr(self, name, eval(raw_param))
-        TEST_CASES.append(self)
+        TEST_CASES.append(self if not xfail else xfail_param(self, reason="test case not yet supported"))
```

Comment:
nice! you might even let `xfail` hold the string reason, so you could convert...